packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -223,138 +256,199 @@ class SyncWatcher extends BaseWatcher {
+        return await this.handleTransferRootConfirmedEvent(event)
+      },
+      this.getSyncOptions(l1Bridge.TransferRootConfirmed)
+    )
+  }
 
-      promises.push(
-        l1Bridge.mapTransferRootConfirmedEvents(
-          async (event: TransferRootConfirmedEvent) => {
-            return await this.handleTransferRootConfirmedEvent(event)
-          },
-          getOptions(l1Bridge.TransferRootConfirmed)
-        )
-      )
+  getTransferBondChallengedEventPromise (): Promise<any> {
+    if (!this.isL1) {
+      return Promise.resolve()
+    }
+    const l1Bridge = this.bridge as L1Bridge
+    return l1Bridge.mapTransferBondChallengedEvents(
+      async (event: TransferBondChallengedEvent) => {
+        return await this.handleTransferBondChallengedEvent(event)
+      },
+      this.getSyncOptions(l1Bridge.TransferBondChallenged)
+    )
+  }
 
-      promises.push(
-        l1Bridge.mapTransferBondChallengedEvents(
-          async (event: TransferBondChallengedEvent) => {
-            return await this.handleTransferBondChallengedEvent(event)
-          },
-          getOptions(l1Bridge.TransferBondChallenged)
-        )
-      )
+  getTransferSentEventPromise (): Promise<any> {
+    if (this.isL1) {
+      return Promise.resolve()
     }
+    const l2Bridge = this.bridge as L2Bridge
+    return l2Bridge.mapTransferSentEvents(
+      async (event: TransferSentEvent) => {
+        return await this.handleTransferSentEvent(event)
+      },
+      this.getSyncOptions(l2Bridge.TransferSent)
+    )
+  }
 
+  getTransferRootSetEventPromise (): Promise<any> {
+    return this.bridge.mapTransferRootSetEvents(
+      async (event: TransferRootSetEvent) => {
+        return await this.handleTransferRootSetEvent(event)
+      },
+      this.getSyncOptions(this.bridge.TransferRootSet)
+    )
+  }
+
+  getTransferRootBondedEventPromise (): Promise<any>{
     if (!this.isL1) {
-      const l2Bridge = this.bridge as L2Bridge
-      promises.push(
-        l2Bridge.mapTransferSentEvents(
-          async (event: TransferSentEvent) => {
-            return await this.handleTransferSentEvent(event)
-          },
-          getOptions(l2Bridge.TransferSent)
-        )
-      )
+      return Promise.resolve()
+    }
+    const l1Bridge = this.bridge as L1Bridge
+    return l1Bridge.mapTransferRootBondedEvents(
+      async (event: TransferRootBondedEvent) => {
+        return await this.handleTransferRootBondedEvent(event)
+      },
+      this.getSyncOptions(l1Bridge.TransferRootBonded)
+    )
+  }
 
-      transferRootInitialEventPromises.push(
-        l2Bridge.mapTransfersCommittedEvents(
-          async (event: TransfersCommittedEvent) => {
-            return await Promise.all([
-              this.handleTransfersCommittedEvent(event)
-            ])
-          },
-          getOptions(l2Bridge.TransfersCommitted)
-        )
-      )
+  getTransfersCommittedEventPromise (): Promise<any> {
+    if (this.isL1) {
+      return Promise.resolve()
```

Comment:
suggestion: add `async` to this function and return empty array instead of `Promise.resolve()`


```suggestion
      return []
```

Reply:
For all of these, the intention was to mimic the current array-of-non-awaited-promises (like [this](https://github.com/hop-protocol/hop/blob/develop/packages/hop-node/src/watchers/SyncWatcher.ts#L253C7-L269)). I'm not 100% sure it is correct though. Since all of these get executed `Promise.all()` do you know what would happen if async was added?

Reply:
Functionally there is no difference; the event map function (ie `getTransfersCommittedEventPromise`) are async functions so returning those in a function makes the function a promise, and adding `async` also makes the function a promise and anything returned by the function is always promisifed. 

Additionally the `await` in the `async event =>` function from old code in some places isn't functionally doing anything so it can be removed as well, e.g.

```ts
async getTransferSentToL2EventPromise(): Promise<any> {
  if (!this.isL1) return [];

  const l1Bridge = this.bridge as L1Bridge;
  return l1Bridge.mapTransferSentToL2Events(
    event => this.handleTransferSentToL2Event(event),
    this.getSyncOptions(l1Bridge.TransferSentToL2)
  );
}
```

Reply:
Thanks for the explanation, makes a ton of sense and I now understand better what it is doing.

IIUC, the empty return still needs to be a `Promise.resovle()` since the return statement expects a promise. For that reason, I left that piece of the code the same. LMK if you have any other thoughts around that. 

Resolved with b6f11e9f9c4f906baa78361f9fd97cfba9820293

Reply:
My comment about the return statement was wrong. I now agree with your whole suggestion. Updated with 39a826ed09925d2327b63cfb629a3a523cb8bd37